MyToolsYourToolsClient/src/app/user-profile/opinions/opinions.component.ts

```diff
@@ -18,12 +19,15 @@ export class OpinionsComponent implements OnInit {
   constructor(private opinionService: OpinionService, private userService: UserService) { }
 
   ngOnInit() {
-  this.opinionService.getOpinions().subscribe(o => this.opinions = o);
-  this.userService.getUsers().subscribe(u => this.users = u);
-  this.opinions = this.opinions.filter( o => o.ratedUserId === this.user.id);
+    this.userService.getUsers().subscribe(u => this.users = u);
+    let allOpinions: Opinion[];
+    this.opinionService.getOpinions()
+      .pipe(
+        tap(allOpinions => this.opinions = allOpinions.filter( o => o.ratedUserId === this.user.id))
```

Comment:
Ogólnie do kontrolera dodałbym funkcję typu GetUserOpinions(userId), żeby za każdym razem nie ściągać na front wszystkich opinii i filtrować ich na froncie, tylko przefiltrowane już zwrócić z backendu, ale na razie możemy to zostawić, bo działa dobrze

Reply:
Ja bartek nie weim co ty masz z tym pipe haha.

Reply:
No właśnie patrzałam jak ty to masz i robiłam podobnie to dlatego, ale to usunę to jak jest niepotrzebne